src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -23,7 +47,14 @@ export const TodoFilter = () => (
 
       <span className="icon is-right" style={{ pointerEvents: 'all' }}>
         {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
+        {isShowClearButton && (
+          <button
+            data-cy="clearSearchButton"
+            onClick={handleCleanSearch}
```

Comment:
Consider adding `event.preventDefault()` inside the `handleCleanSearch` function to prevent the default button behavior, which could cause a page reload or form submission.